src/solvers/refinement/string_refinement.cpp

```diff
@@ -657,7 +654,11 @@ decision_proceduret::resultt string_refinementt::dec_solve()
   std::vector<equal_exprt> local_equations;
   for(const equal_exprt &eq : equations)
   {
-    if(!add_node(dependencies, eq, generator.array_pool))
+    equal_exprt eq_copy = eq;
+    // Char array symbols are replaced by cannonical element to ensure
+    // equal arrays are associated to the same nodes in the graph.
+    symbol_resolve.replace_expr(eq_copy);
```

Comment:
I think it would be nice to extract that into a separate function for the following reasons:
* It makes it clear, without the use of comments, that this is a single operation, and that `eq_copy` goes out of scope immediately after the operation
* It gives an opportunity to document what we are doing here. It's not obvious and I'm worried we'll have trouble understanding what we're doing once the commit message is gone.

Plus, I imagine this mechanism should will potentially also be used elsewhere.

Reply:
@joel I'm not sure which part you'd like to extract, just that instruction, what's inside the loop or the full loop?

Reply:
I've added a commit `Use transform_expr to clarify loop in dec_solve` which should make that a bit clearer but I don't think it's exactly what you had in mind

Reply:
That's much clearer. Thanks!